client/html/src/Client/Html/Checkout/Standard/Process/Standard.php

```diff
@@ -266,15 +270,25 @@ public function process()
 			$orderCntl = \Aimeos\Controller\Frontend\Factory::createController( $context, 'order' );
 			$basketCntl = \Aimeos\Controller\Frontend\Factory::createController( $context, 'basket' );
 
-			$basketCntl->get()->setCustomerId( $context->getUserId() );
 
-			parent::process();
+			if ( $view->param( 'cs_order', null ) !== null ) {
```

Comment:
But the payment provider is called independently of the new if-block.

The if-block creates the order if cs_order is set, and the else-if-block loads it from the database. In both cases the payment provider is called afterwards.

Creating the order when cp_payment is set may still be a better option (instead of creating it on cs_order and loading it from the database on cp_payment), but I don't think that is what you were saying.